core/simulator.h

```diff
@@ -31,22 +32,33 @@ class Simulator : public QObject {
   void stopped();
 
  public slots:
+  // Responds to control flow signals from the GUI and scripts. Start, stop, and
+  // step are self-explanatory. stepForParticleAt executes one activation for
+  // the specific particle at the given node. setStepDuration updates the delay
+  // in milliseconds between particle activations. runUntilTermination activates
+  // particles repeatedly until the hasTerminated condition is satisfied.
   void start();
   void stop();
   void step();
   void stepForParticleAt(Node node);
   void setStepDuration(int ms);
   void runUntilTermination();
-  void saveScreenshotSetup(const QString filePath);
 
+  // Responds to GUI and script requests for statistics and metrics.
   int numParticles() const;
   int numObjects() const;
-  int numMovements() const;
-  int numRounds() const;
+  QVariant metrics() const;
+
+  // Responds to the exportMetrics signal from the GUI and scripts by creating
+  // an output file with a unique timestamp (to avoid accidental overwrites) and
+  // writing the metrics JSON to it.
+  void exportMetrics();
+
+  // TODO.
```

Comment:
Did you want to keep the comment? Or should a GitHub issue be made for implementing this part (so the comment + function can be moved to a separate PR)?

Reply:
All the functions here were already implemented; it's just that `simulator.h` didn't have any header function comments. I've added one for `saveScreenshotSetup` in the most recent commit.